PCF-ADR-0002-release-story-and-branch-maintenance.md

```diff
@@ -0,0 +1,143 @@
+these principles:
+
+- The `main` branch is always deployable and represents the latest version of the code.
+- New features and bug fixes are developed in short-lived branches, created from the `main` branch.
+- When a branch is ready for review, a pull request is opened to merge it into the `main` branch.
+- After the pull request has both an `approved` and `lgtm` label, it is merged into the `main` branch and all necessary
+images and artifacts are pushed to indexes where they can be used for deployment in dev/latest environments.
+- If a bugfix is required for any release, the bugfix is added using the above process and then cherry-picked into the
+  branch for the minor release. (explained below)
+
+### Rationale
+
+- It is simple and easy to understand and follow. There is no need to maintain multiple long-lived branches or deal with
+  complex merge conflicts.
+- It supports continuous integration and delivery. Every commit on the `main` branch is tested and deployable, ensuring
+  high quality and fast feedback.
+- It encourages collaboration and code review. Pull requests are the central place for discussing and reviewing code
+  changes, ensuring that every change is approved by at least one other developer before merging.
+- It allows us to release frequently and incrementally. We can deploy new features and bug fixes as soon as they are
+  ready, without waiting for a predefined release cycle or schedule.
+
+### How to pull in bug fixes into a release branch
+
+Sometimes, we may need to create a release branch from the `main` branch to prepare for a major or minor release. For
+example, we may want to freeze the feature set for a certain version and only allow bug fixes on that branch. In that
+case, we can use the following steps to pull in bug fixes from the `main` branch into the release branch:
+
+- Create a release branch from the `main` branch with a name that indicates the minor version number, such as `v1.2` or 
+  `release-v1.2`.
+- Continue developing new features and bug fixes on the `main` branch as usual.
+- If a bug fix on the `main` branch needs to be included in the release branch, cherry-pick the commit that contains the
+  fix onto the release branch.
+- When the branch is ready for a new release, create a tag on GitHub with the patch version incremented. For example if
+  `v1.2.0` is the last released version for `v1.2`, then the new tag will be `v1.2.1`. This will trigger the release
+  pipeline for that tag.
+
+### Release process
+
+The release process for our project is very simple and consists of creating a version tag on GitHub. For example, if we
+want to release version 1.3 of our software, we just need to create a tag named `v1.3.0` on GitHub from the latest
+commit on the `main` branch. This will trigger the build and release pipelines, or someone will manually take the
+required steps for generating build artifacts, etc..
+
```

Comment:
We may want to store the artifacts on a stagging area for a few days for people to download and validate as a last gate before going out.

Reply:
I think this is definitely doable. One could create `v1.1.0-alpha` tag and then pull any bug fixes that are made into the `v1.1` branch as normal. The second tag on that branch would be `v1.1.0` or even `v1.0.0-beta`. I'll add this to the dev process as an option for staging releases.

Reply:
Looks good. Well explained.